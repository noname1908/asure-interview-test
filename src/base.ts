import {
  Stack,
  StackProps,
  aws_ec2 as ec2,
  aws_elasticloadbalancingv2 as elbv2,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { config } from './config';
import { ApplicationLoadBalancer } from './constructs/alb';
import { CustomVPC } from './constructs/vpc';

export class BaseStack extends Stack {
  public readonly vpc: ec2.IVpc;
  public readonly albListener: elbv2.IApplicationListener;
  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

    // VPC
    const { vpc } = new CustomVPC(this, { prefix: config.projectName });
    this.vpc = vpc;

    // ALB
    const { listener } = new ApplicationLoadBalancer(this, {
      prefix: config.projectName,
      vpc,
    });
    this.albListener = listener;
  }
}
