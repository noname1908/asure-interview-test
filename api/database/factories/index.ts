// import Factory from '@ioc:Adonis/Lucid/Factory'

export { default as RoleFactory } from './RoleFactory'
export { default as UserFactory } from './UserFactory'
