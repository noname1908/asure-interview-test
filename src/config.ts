import * as dotenv from 'dotenv';
// load environments from .env
dotenv.config();

const stage = process.env.STAGE || 'dev';

export const config = {
  // this prefix is used to name provisioned resources
  projectName: `asure-interview-test-${stage}`,
  stage,
  env: {
    account: process.env.AWS_ACCOUNT,
    region: process.env.AWS_REGION || 'ap-southeast-1',
  },
};
