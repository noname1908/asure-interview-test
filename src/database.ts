import { Stack, StackProps, aws_ec2 as ec2 } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { config } from './config';
import { MySQLRdsInstance } from './constructs/rds';

interface RdsMysqlStackProps extends StackProps {
  vpc: ec2.IVpc;
}

export class RdsMysqlStack extends Stack {
  public readonly dbSecretName: string;

  constructor(scope: Construct, id: string, props: RdsMysqlStackProps) {
    super(scope, id, props);

    const vpc = props.vpc;

    // RDS mysql database
    const mysqlRds = new MySQLRdsInstance(this, {
      prefix: config.projectName,
      vpc,
      username: 'admin',
      database: 'testing',
      port: 3306,
    });

    this.dbSecretName = mysqlRds.dbSecretName;
  }
}
