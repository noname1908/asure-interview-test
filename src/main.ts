import { execSync, ExecSyncOptions } from 'child_process';
import path from 'path';
import {
  App,
  Stack,
  StackProps,
  Duration,
  DockerImage,
  aws_s3_assets as s3assets,
  aws_ec2 as ec2,
  aws_elasticloadbalancingv2 as elbv2,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { copySync } from 'fs-extra';
import { BaseStack } from './base';
import { config } from './config';
import { EC2AutoScalingGroup } from './constructs/ec2';
import { RdsMysqlStack } from './database';

interface EC2FullStackProps extends StackProps {
  vpc: ec2.IVpc;
  albListener: elbv2.IApplicationListener;
  dbSecretPath: string;
}

export class EC2FullStack extends Stack {
  constructor(scope: Construct, id: string, props: EC2FullStackProps) {
    super(scope, id, props);

    const vpc = props.vpc;
    const listener = props.albListener;
    const dbSecretPath = props.dbSecretPath;

    // EC2 instance in an autoscaling group
    const { asg } = new EC2AutoScalingGroup(this, {
      prefix: config.projectName,
      vpc,
    });

    // add the autoscaling group to the load balancer
    listener.addTargets(`${config.projectName}-asg-targets`, {
      port: 80,
      targets: [asg],
      healthCheck: {
        path: '/',
        unhealthyThresholdCount: 2,
        healthyThresholdCount: 5,
        interval: Duration.seconds(30),
      },
    });

    // add scaling policy for the auto scaling group
    asg.scaleOnRequestCount('request-per-minute-scaling', {
      targetRequestsPerMinute: 100,
    });
    asg.scaleOnCpuUtilization('cpu-scaling', {
      targetUtilizationPercent: 70,
    });

    // --- App ---
    const execOptions: ExecSyncOptions = { stdio: 'inherit' };

    // --- API ---
    // Upload the api to s3
    const apiAsset = new s3assets.Asset(this, 'ApiAsset', {
      path: path.join(__dirname, '../api'),
      bundling: {
        image: DockerImage.fromRegistry('alpine'),
        local: {
          tryBundle(outputDir: string) {
            try {
              execSync(
                'cd api && yarn install --frozen-lockfile && yarn build && cp .env.example ./build/.env',
                execOptions,
              );
              copySync('./api/build', outputDir, {
                ...execOptions,
                recursive: true,
              });
              return true;
            } catch (error) {
              return false;
            }
          },
        },
      },
    });

    // Allow EC2 instance to read the files from the api s3 bucket
    apiAsset.grantRead(asg.role);

    // Download the api files from S3
    const apiFilePath = asg.userData.addS3DownloadCommand({
      bucket: apiAsset.bucket,
      bucketKey: apiAsset.s3ObjectKey,
    });
    // #--- API ---

    // --- Site
    // Upload the site to s3
    const siteAsset = new s3assets.Asset(this, 'SiteAsset', {
      path: path.join(__dirname, '../site'),
      bundling: {
        image: DockerImage.fromRegistry('alpine'),
        local: {
          tryBundle(outputDir: string) {
            try {
              execSync(
                'cd site && yarn install --frozen-lockfile && yarn generate && cp ../nginx/nginx-app.conf ./dist/nginx-app.conf',
                execOptions,
              );
              copySync('./site/dist', outputDir, {
                ...execOptions,
                recursive: true,
              });
              return true;
            } catch (error) {
              return false;
            }
          },
        },
      },
    });

    // Allow EC2 instance to read the files
    siteAsset.grantRead(asg.role);

    // Download the file from S3
    const siteFilePath = asg.userData.addS3DownloadCommand({
      bucket: siteAsset.bucket,
      bucketKey: siteAsset.s3ObjectKey,
    });
    // #--- Site ---
    // #--- App ---

    // --- Configuration Script ---

    // Upload the configuration file to s3
    const configScriptAsset = new s3assets.Asset(this, 'ConfigScriptAsset', {
      path: path.join(__dirname, '../scripts/configure_app.sh'),
    });

    // Allow EC2 instance to read the file
    configScriptAsset.grantRead(asg.role);

    // Download the file from s3
    const configScriptFilePath = asg.userData.addS3DownloadCommand({
      bucket: configScriptAsset.bucket,
      bucketKey: configScriptAsset.s3ObjectKey,
    });

    // Execute the downloaded file
    asg.userData.addExecuteFileCommand({
      filePath: configScriptFilePath,
      arguments: [
        siteFilePath,
        apiFilePath,
        config.env.region,
        dbSecretPath,
      ].join(' '),
    });
    // #--- Configuration Script ---
  }
}

const app = new App();
// base
const base = new BaseStack(app, `${config.projectName}-base`, {
  env: config.env,
  description: 'Deploys base infrastructure',
  tags: { Project: config.projectName },
});
// mysql
const database = new RdsMysqlStack(app, `${config.projectName}-rds-mysql`, {
  env: config.env,
  description: 'Deploys database infrastructure',
  tags: { Project: config.projectName },
  vpc: base.vpc,
});
// ec2
new EC2FullStack(app, `${config.projectName}-ec2`, {
  env: config.env,
  description: 'Deploys EC2 scaling group for web application infrastructure',
  tags: { Project: config.projectName },
  vpc: base.vpc,
  albListener: base.albListener,
  dbSecretPath: database.dbSecretName,
});

app.synth();
