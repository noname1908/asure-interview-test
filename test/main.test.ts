import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { BaseStack } from '../src/base';
import { config } from '../src/config';
import { RdsMysqlStack } from '../src/database';
import { EC2FullStack } from '../src/main';

test('Snapshot', () => {
  const app = new App();
  const base = new BaseStack(app, 'test-base', {
    env: config.env,
  });
  const database = new RdsMysqlStack(app, 'test-rds-mysql', {
    env: config.env,
    description: 'Deploys database infrastructure',
    tags: { Project: config.projectName },
    vpc: base.vpc,
  });
  const stack = new EC2FullStack(app, 'test-ec2', {
    env: config.env,
    vpc: base.vpc,
    albListener: base.albListener,
    dbSecretPath: database.dbSecretName,
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
