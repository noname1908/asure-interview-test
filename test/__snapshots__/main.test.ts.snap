// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueawsserviceamiamazonlinuxlatestamzn2amihvmx8664gp2C96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
    },
  },
  "Resources": {
    "asureinterviewtestdevasgASG6E502ADE": {
      "Properties": {
        "LaunchConfigurationName": {
          "Ref": "asureinterviewtestdevasgLaunchConfigA0CA8CF7",
        },
        "MaxSize": "2",
        "MinSize": "1",
        "Tags": [
          {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "test-ec2/asure-interview-test-dev-asg",
          },
        ],
        "TargetGroupARNs": [
          {
            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevalbasureinterviewtestdevalblistenerasureinterviewtestdevasgtargetsGroup2FB779FDDFC01BC6",
          },
        ],
        "VPCZoneIdentifier": [
          {
            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevvpcPrivateSubnet1Subnet28B6093056B332DB",
          },
          {
            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevvpcPrivateSubnet2Subnet8267CFEA7014593B",
          },
          {
            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevvpcPrivateSubnet3Subnet88596C49E6FFCB6E",
          },
        ],
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
      "UpdatePolicy": {
        "AutoScalingScheduledAction": {
          "IgnoreUnmodifiedGroupSizeProperties": true,
        },
      },
    },
    "asureinterviewtestdevasgInstanceProfile58FB736E": {
      "Properties": {
        "Roles": [
          {
            "Ref": "asureinterviewtestdevec2role53D60150",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "asureinterviewtestdevasgInstanceSecurityGroupEF8F0EBB": {
      "Properties": {
        "GroupDescription": "test-ec2/asure-interview-test-dev-asg/InstanceSecurityGroup",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "Tags": [
          {
            "Key": "Name",
            "Value": "test-ec2/asure-interview-test-dev-asg",
          },
        ],
        "VpcId": {
          "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevvpc03830C289090CB34",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "asureinterviewtestdevasgInstanceSecurityGroupfromtestbaseasureinterviewtestdevalbSecurityGroup3E515B9E805FC1C58F": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 80,
        "GroupId": {
          "Fn::GetAtt": [
            "asureinterviewtestdevasgInstanceSecurityGroupEF8F0EBB",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::ImportValue": "test-base:ExportsOutputFnGetAttasureinterviewtestdevalbSecurityGroup5EB71584GroupId2BC4EFDF",
        },
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "asureinterviewtestdevasgInstanceSecurityGrouptestbaseasureinterviewtestdevalbSecurityGroup3E515B9E80from0CF4A169": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "asureinterviewtestdevasgInstanceSecurityGroupEF8F0EBB",
            "GroupId",
          ],
        },
        "FromPort": 80,
        "GroupId": {
          "Fn::ImportValue": "test-base:ExportsOutputFnGetAttasureinterviewtestdevalbSecurityGroup5EB71584GroupId2BC4EFDF",
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "asureinterviewtestdevasgLaunchConfigA0CA8CF7": {
      "DependsOn": [
        "asureinterviewtestdevec2roleDefaultPolicy14EF8A29",
        "asureinterviewtestdevec2role53D60150",
      ],
      "Properties": {
        "IamInstanceProfile": {
          "Ref": "asureinterviewtestdevasgInstanceProfile58FB736E",
        },
        "ImageId": {
          "Ref": "SsmParameterValueawsserviceamiamazonlinuxlatestamzn2amihvmx8664gp2C96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "InstanceType": "t2.micro",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "asureinterviewtestdevasgInstanceSecurityGroupEF8F0EBB",
              "GroupId",
            ],
          },
        ],
        "UserData": {
          "Fn::Base64": {
            "Fn::Join": [
              "",
              [
                "#!/bin/bash
mkdir -p $(dirname '/tmp/0362b338d6c386445d145c84715ebceef7f333d63d20ff6f5707db96f3bd5f89.zip')
aws s3 cp 's3://cdk-hnb659fds-assets-242729241172-ap-southeast-1/0362b338d6c386445d145c84715ebceef7f333d63d20ff6f5707db96f3bd5f89.zip' '/tmp/0362b338d6c386445d145c84715ebceef7f333d63d20ff6f5707db96f3bd5f89.zip'
mkdir -p $(dirname '/tmp/16a37b3155cd45adfb146b49f91ab55806a1ee9f24751482b7e2492697c1fe4e.zip')
aws s3 cp 's3://cdk-hnb659fds-assets-242729241172-ap-southeast-1/16a37b3155cd45adfb146b49f91ab55806a1ee9f24751482b7e2492697c1fe4e.zip' '/tmp/16a37b3155cd45adfb146b49f91ab55806a1ee9f24751482b7e2492697c1fe4e.zip'
mkdir -p $(dirname '/tmp/78b3c1ae8b6d28f80071ed436c023440c25945a15de6988f9fe028b658d80700.sh')
aws s3 cp 's3://cdk-hnb659fds-assets-242729241172-ap-southeast-1/78b3c1ae8b6d28f80071ed436c023440c25945a15de6988f9fe028b658d80700.sh' '/tmp/78b3c1ae8b6d28f80071ed436c023440c25945a15de6988f9fe028b658d80700.sh'
set -e
chmod +x '/tmp/78b3c1ae8b6d28f80071ed436c023440c25945a15de6988f9fe028b658d80700.sh'
'/tmp/78b3c1ae8b6d28f80071ed436c023440c25945a15de6988f9fe028b658d80700.sh' /tmp/16a37b3155cd45adfb146b49f91ab55806a1ee9f24751482b7e2492697c1fe4e.zip /tmp/0362b338d6c386445d145c84715ebceef7f333d63d20ff6f5707db96f3bd5f89.zip ap-southeast-1 ",
                {
                  "Fn::Join": [
                    "-",
                    [
                      {
                        "Fn::Select": [
                          0,
                          {
                            "Fn::Split": [
                              "-",
                              {
                                "Fn::Select": [
                                  6,
                                  {
                                    "Fn::Split": [
                                      ":",
                                      {
                                        "Fn::ImportValue": "test-rds-mysql:ExportsOutputRefasureinterviewtestdevmysqlcredentialsAC28DCC9B76B4DEE",
                                      },
                                    ],
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                      {
                        "Fn::Select": [
                          1,
                          {
                            "Fn::Split": [
                              "-",
                              {
                                "Fn::Select": [
                                  6,
                                  {
                                    "Fn::Split": [
                                      ":",
                                      {
                                        "Fn::ImportValue": "test-rds-mysql:ExportsOutputRefasureinterviewtestdevmysqlcredentialsAC28DCC9B76B4DEE",
                                      },
                                    ],
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                      {
                        "Fn::Select": [
                          2,
                          {
                            "Fn::Split": [
                              "-",
                              {
                                "Fn::Select": [
                                  6,
                                  {
                                    "Fn::Split": [
                                      ":",
                                      {
                                        "Fn::ImportValue": "test-rds-mysql:ExportsOutputRefasureinterviewtestdevmysqlcredentialsAC28DCC9B76B4DEE",
                                      },
                                    ],
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                      {
                        "Fn::Select": [
                          3,
                          {
                            "Fn::Split": [
                              "-",
                              {
                                "Fn::Select": [
                                  6,
                                  {
                                    "Fn::Split": [
                                      ":",
                                      {
                                        "Fn::ImportValue": "test-rds-mysql:ExportsOutputRefasureinterviewtestdevmysqlcredentialsAC28DCC9B76B4DEE",
                                      },
                                    ],
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  ],
                },
              ],
            ],
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "asureinterviewtestdevasgScalingPolicycpuscaling29B0774D": {
      "Properties": {
        "AutoScalingGroupName": {
          "Ref": "asureinterviewtestdevasgASG6E502ADE",
        },
        "PolicyType": "TargetTrackingScaling",
        "TargetTrackingConfiguration": {
          "PredefinedMetricSpecification": {
            "PredefinedMetricType": "ASGAverageCPUUtilization",
          },
          "TargetValue": 70,
        },
      },
      "Type": "AWS::AutoScaling::ScalingPolicy",
    },
    "asureinterviewtestdevasgScalingPolicyrequestperminutescalingE83D377B": {
      "Properties": {
        "AutoScalingGroupName": {
          "Ref": "asureinterviewtestdevasgASG6E502ADE",
        },
        "PolicyType": "TargetTrackingScaling",
        "TargetTrackingConfiguration": {
          "PredefinedMetricSpecification": {
            "PredefinedMetricType": "ALBRequestCountPerTarget",
            "ResourceLabel": {
              "Fn::Join": [
                "",
                [
                  {
                    "Fn::Select": [
                      1,
                      {
                        "Fn::Split": [
                          "/",
                          {
                            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevalbasureinterviewtestdevalblistenerE86B9EEE5B3E543D",
                          },
                        ],
                      },
                    ],
                  },
                  "/",
                  {
                    "Fn::Select": [
                      2,
                      {
                        "Fn::Split": [
                          "/",
                          {
                            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevalbasureinterviewtestdevalblistenerE86B9EEE5B3E543D",
                          },
                        ],
                      },
                    ],
                  },
                  "/",
                  {
                    "Fn::Select": [
                      3,
                      {
                        "Fn::Split": [
                          "/",
                          {
                            "Fn::ImportValue": "test-base:ExportsOutputRefasureinterviewtestdevalbasureinterviewtestdevalblistenerE86B9EEE5B3E543D",
                          },
                        ],
                      },
                    ],
                  },
                  "/",
                  {
                    "Fn::ImportValue": "test-base:ExportsOutputFnGetAttasureinterviewtestdevalbasureinterviewtestdevalblistenerasureinterviewtestdevasgtargetsGroup2FB779FDTargetGroupFullName0CF54918",
                  },
                ],
              ],
            },
          },
          "TargetValue": 100,
        },
      },
      "Type": "AWS::AutoScaling::ScalingPolicy",
    },
    "asureinterviewtestdevec2role53D60150": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/SecretsManagerReadWrite",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "asureinterviewtestdevec2roleDefaultPolicy14EF8A29": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-242729241172-ap-southeast-1",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-242729241172-ap-southeast-1/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "asureinterviewtestdevec2roleDefaultPolicy14EF8A29",
        "Roles": [
          {
            "Ref": "asureinterviewtestdevec2role53D60150",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
